src/cache/file_cache.cc

```diff
@@ -27,6 +27,25 @@ String ReplaceTildeInPath(const String& path) {
   return path;
 }
 
+String CombineHashes(const Immutable& first, const Immutable& second) {
+  if (second.empty()) {
+    return first;
+  }
+  return base::Hexify((first + "-"_l + second).Hash());
+}
+
+String HashCombine(const List<Immutable>& files) {
```

Comment:
```
String HashCombine(const List<Immutable>& files) {
  String hashes_string;
  for (auto it = files.begin(); it != files.end(); ++it) {
    if (it != files.begin()) {
      hashes_string += "-";
    }
    hashes_string += base::Hexify(it->Hash());
  }
  return base::Hexify(Immutable(hashes_string).Hash());
}
```

Probably comparing iterator in the input sequence would be easier to inderstand than thinking about empty `hashes_string`.